contracts/accumulator/types/Accumulator6.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.13;
+
+import "../../number/types/Fixed6.sol";
+import "../../number/types/UFixed6.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+
+/// @dev Accumulator6 type
+struct Accumulator6 {
+    Fixed6 _value;
+}
+
+using Accumulator6Lib for Accumulator6 global;
+struct StoredAccumulator6 {
+    int256 _value;
+}
+struct Accumulator6Storage { StoredAccumulator6 value; }
+using Accumulator6StorageLib for Accumulator6Storage global;
+
+
+/**
+ * @title Accumulator6Lib
+ * @notice Library that surfaces math operations for the signed Accumulator type.
+ * @dev TODO
+ */
+library Accumulator6Lib {
+    // TODO: Natspec
+    function accumulated(Accumulator6 memory self, Accumulator6 memory from, UFixed6 total) internal pure returns (Fixed6) {
+        return _mul(self._value.sub(from._value), total);
+    }
+
+    /**
+     * @notice Increments an accumulator by a given ratio
+     * @dev Always rounds down in order to prevent overstating the accumulated value
+     * @param self The accumulator to increment
+     * @param amount Numerator of the ratio
+     * @param total Denominator of the ratio
+     */
+    function increment(Accumulator6 memory self, Fixed6 amount, UFixed6 total) internal pure {
+        self._value = self._value.add(_div(amount, total));
+    }
+
+    /**
+     * @notice Decrements an accumulator by a given ratio
+     * @dev Always rounds down in order to prevent overstating the accumulated value
+     * @param self The accumulator to decrement
+     * @param amount Numerator of the ratio
+     * @param total Denominator of the ratio
+     */
+    function decrement(Accumulator6 memory self, Fixed6 amount, UFixed6 total) internal pure {
+        self._value = self._value.add(_div(amount.mul(Fixed6Lib.NEG_ONE), total));
+    }
+
+    function _div(Fixed6 amount, UFixed6 total) private pure returns (Fixed6) {
+        if (amount.sign() == -1) {
+            return Fixed6Lib.from(-1, amount.abs().divOut(total));
```

Comment:
For this I think you can just do:

`return amount.divOut(Fixed6Lib.from(total));`

Could probably shorten this then to something like:

```
return amount.sign() == -1 ? amount.divOut(Fixed6Lib.from(total)) : amount.div(Fixed6Lib.from(total));
```